test/unit/mongo_logger.test.ts

```diff
@@ -0,0 +1,347 @@
+  });
+});
+
+describe('class MongoLogger', function () {
+  describe('#constructor()', function () {
+    it('assigns each property from the options object onto the logging class', function () {
+      const componentSeverities: MongoLoggerOptions['componentSeverities'] = {
+        command: 'alert'
+      } as any;
+      const stream = new Writable();
+      const logger = new MongoLogger({
+        componentSeverities,
+        maxDocumentLength: 10,
+        logDestination: stream
+      });
+
+      expect(logger).to.have.property('componentSeverities', componentSeverities);
+      expect(logger).to.have.property('maxDocumentLength', 10);
+      expect(logger).to.have.property('logDestination', stream);
+    });
+  });
+
+  describe('static #resolveOptions()', function () {
+    describe('componentSeverities', function () {
+      const components = [
+        'MONGODB_LOG_COMMAND',
+        'MONGODB_LOG_TOPOLOGY',
+        'MONGODB_LOG_SERVER_SELECTION',
+        'MONGODB_LOG_CONNECTION'
+      ];
+      const mapToInternalRepresentation = (component: string) => {
+        const options: Record<string, string> = {
+          MONGODB_LOG_COMMAND: 'command',
+          MONGODB_LOG_TOPOLOGY: 'topology',
+          MONGODB_LOG_SERVER_SELECTION: 'serverSelection',
+          MONGODB_LOG_CONNECTION: 'connection'
+        };
+        return options[component];
+      };
+
+      context('MONGODB_LOG_ALL', () => {
+        context('when a default is provided', () => {
+          it('sets default to the provided value', () => {
+            const options = MongoLogger.resolveOptions(
+              { MONGODB_LOG_ALL: SeverityLevel.ALERT },
+              {}
+            );
+            expect(options.componentSeverities).to.have.property('default', SeverityLevel.ALERT);
+          });
+        });
+        context('when no value is provided', () => {
+          it('sets default to off', () => {
+            const options = MongoLogger.resolveOptions({ MONGODB_LOG_ALL: SeverityLevel.OFF }, {});
+            expect(options.componentSeverities).to.have.property('default', SeverityLevel.OFF);
+          });
+        });
+
+        it('is case insensitive', () => {
+          const options = MongoLogger.resolveOptions({ MONGODB_LOG_ALL: 'dEbUg' }, {});
+          expect(options.componentSeverities).to.have.property('default', SeverityLevel.DEBUG);
+        });
+      });
+
+      for (const component of components) {
+        const mappedComponent = mapToInternalRepresentation(component);
+        context(`${component}`, function () {
+          context(`when set to a valid value in the environment`, function () {
+            context('when there is a default provided', function () {
+              it(`sets ${mappedComponent} to the provided value and ignores the default`, function () {
+                const options = MongoLogger.resolveOptions(
+                  { [component]: SeverityLevel.ALERT, MONGODB_LOG_ALL: SeverityLevel.OFF },
```

Comment:
the default should be something other than `OFF` for this case, and then we need a separate case for when the default is `OFF`